packages/database/src/migrations/20210922211939-LESMIS-number-of-schools-overlays-modifies-data.js

```diff
@@ -0,0 +1,324 @@
+  id: reportCode,
+  report_code: reportCode,
+  legacy: false,
+  name,
+  userGroup: PERMISSION_GROUP,
+  dataElementCode: 'value',
+  isDataRegional: true,
+  measureBuilder: 'useReportServer',
+  measureBuilderConfig: {
+    dataSourceType: 'custom',
+    reportCode,
+  },
+  presentationOptions: {
+    scaleType: 'neutral',
+    displayType: 'shaded-spectrum',
+    measureLevel: reportCode.includes('Province') ? 'District' : 'SubDistrict',
+    valueType: 'number',
+    scaleBounds: {
+      left: {
+        max: '0',
+      },
+      right: {
+        min: 'auto',
+      },
+    },
+    periodGranularity: 'one_year_at_a_time',
+  },
+  countryCodes: '{"LA"}',
+  projectCodes: '{laos_schools}',
+});
+
+const addMapOverlayGroup = async (db, parentCode, { name, code }) => {
+  const parentId = await codeToId(db, 'map_overlay_group', parentCode);
+
+  const overlayGroupId = generateId();
+  await insertObject(db, 'map_overlay_group', { id: overlayGroupId, name, code });
+
+  return insertObject(db, 'map_overlay_group_relation', {
+    id: generateId(),
+    map_overlay_group_id: parentId,
+    child_id: overlayGroupId,
+    child_type: 'mapOverlayGroup',
+  });
+};
+
+const addMapOverlay = async (db, parentCode, config) => {
+  const { reportCode, dataElements, name, sortOrder } = config;
+  const report = getReport(reportCode, dataElements);
+  const mapOverlay = getMapOverlay(name, reportCode);
+  const permissionGroupId = await nameToId(db, 'permission_group', PERMISSION_GROUP);
+  await insertObject(db, 'report', {
+    id: generateId(),
+    permission_group_id: permissionGroupId,
+    ...report,
+  });
+  await insertObject(db, 'mapOverlay', mapOverlay);
+
+  const mapOverlayGroupId = await codeToId(db, 'map_overlay_group', parentCode);
+  return insertObject(db, 'map_overlay_group_relation', {
+    id: generateId(),
+    map_overlay_group_id: mapOverlayGroupId,
+    child_id: mapOverlay.id,
+    child_type: 'mapOverlay',
+    sort_order: sortOrder,
+  });
+};
+
+const OVERLAY_CODE = 'LESMIS_Number_Of_Schools';
+
+const OVERLAY_GROUPS = [
+  {
+    parentCode: 'Root',
+    children: [
+      {
+        name: 'Number Of Schools and Institutions',
+        code: `${OVERLAY_CODE}_Group`,
+      },
+    ],
+  },
+  {
+    parentCode: `${OVERLAY_CODE}_Group`,
+    children: [
+      {
+        name: 'District Level',
+        code: `${OVERLAY_CODE}_District_Group`,
+      },
+      {
+        name: 'Province Level',
+        code: `${OVERLAY_CODE}_Province_Group`,
+      },
+    ],
+  },
+];
+
+const MAP_OVERLAYS = [
+  {
+    parentCode: `${OVERLAY_CODE}_District_Group`,
+    children: [
+      {
+        reportCode: `${OVERLAY_CODE}_District_ECE_map`,
+        name: 'Number of ECE Centres/Kindergartens',
+        dataElements: [
+          'nosch_type1_private',
+          'nosch_type1_public',
+          'nosch_type2_private',
+          'nosch_type2_public',
+          'nosch_type3_private',
+          'nosch_type3_public',
```

Comment:
There are indicators for `nosch_ece` and such, any particular reason you skipped using those?

Reply:
Oh maybe because I didn't split lower and upper secondary in the indicators 🤔

Reply:
no I just didn't see them when I looked up the dataElements. True that it wont work for the `lse` and `use` ones but I might as well use them for the others.